x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java

```diff
@@ -488,6 +491,10 @@ private PhysicalOperation planEnrich(EnrichExec enrich, LocalExecutionPlannerCon
 
     private PhysicalOperation planHashJoin(HashJoinExec join, LocalExecutionPlannerContext context) {
         PhysicalOperation source = plan(join.child(), context);
+        if (join.matchFields().isEmpty()) {
```

Comment:
I think this has a fair chance of becoming very incorrect; at least it's confusing, because it assumes knowledge that
- the only way for empty match fields is from an inlinestats without a by clause (I'm actually not sure omitting the `ON` in LOOKUP is actually forbidden - if/once it isn't, this here is wrong), and
- without a by clause we'll have just 1 row that needs to be appended to every existing row.

Let's add another condition: a check for whether or not we only have 1 row in the hashjoin's local rel., then it's always correct. We perform this check as an assertion, anyway.

Reply:
I'd like to have another look at this one. Rather than mapping HashJoin I'd love to replace the result as an EVAL. I'll bet there's a better way...

Reply:
I did it!